### ISSUE TITLE
Nested fragment results arent't merged into a query result

### ISSUE THREAD

**Issue by amannn** [apollographql/apollo-client#1479]
When a fragment selects nested fields and the original query also selects fields on that type, the fields from the fragment are left out and only the ones from the query are in the result.

**Intended outcome:**

The results from the nested fields of the fragment should be merged into the complete result.

**Actual outcome:**

The nested fields are left out.

**How to reproduce the issue:**

[Here's a query](http://www.githunt.com/graphiql?query=fragment%20RepositoryData%20on%20Repository%20%7B%0A%20%20full_name%0A%20%20owner%20%7B%0A%20%20%20%20login%0A%20%20%7D%0A%7D%0A%0Aquery%20feed%20%7B%0A%20%20feed(type%3A%20NEW%2C%20limit%3A%205)%20%7B%0A%20%20%20%20score%0A%20%20%20%20repository%20%7B%0A%20%20%20%20%20%20...RepositoryData%0A%20%20%20%20%20%20name%0A%20%20%20%20%20%20owner%20%7B%0A%20%20%20%20%20%20%20%20avatar_url%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A&operationName=feed
) that can be run in Githunts' GraphiQL. Notice that `login` as well as `avatar_url` is available in the result.

When the same query is run with apollo-client, only `avatar_url` is available in the result.

Here's a node script that shows the issue:

```js
require('isomorphic-fetch');
const {default: ApolloClient, createNetworkInterface} = require('apollo-client');
const gql = require('graphql-tag').default;

const client = new ApolloClient({
  networkInterface: createNetworkInterface({uri: 'http://www.githunt.com/graphql'})
});

const query = gql`
fragment RepositoryData on Repository {
  owner {
    login
  }
}

{
  feed(type: NEW, limit: 5) {
    repository {
      ...RepositoryData
      name
      owner {
        avatar_url
      }
    }
  }
}
`;

const promise = client.query({query}).then(({data}) => {
  console.log(JSON.stringify(data));
}, e => {
  console.error(res);
});

```

**Comment by helfer** [comment]
@amannn any chance that the `Repository` type is a union or an interface? If so, we're about to fix this in #1483

**Comment by amannn** [comment]
Thanks for the quick reply! [`Repository` is a regular type](https://github.com/apollographql/GitHunt-API/blob/master/api/github/schema.js#L6). Also in my app code I'm encountering this bug with a regular type that is not related to a union or an interface.

Here's a fork of `react-apollo-error-template` that shows the error: https://github.com/amannn/react-apollo-error-template. In this case the [name of the organization](https://github.com/amannn/react-apollo-error-template/blob/master/src/App.js#L29) isn't included in the data that is passed to the view.

As I tried to show with my node example, this error shouldn't be related to `react-apollo` but rather to `apollo-client`.

Thanks for your help!

**Comment by helfer** [comment]
@amannn sorry this dropped off my radar. Are you still facing this issue?

**Comment by amannn** [comment]
@helfer No worries, I was able to work around it so far.

I just updated my node script to use `apollo-client@1.2.0`, `graphql-tag@2.0.0` and `isomorphic-fetch@2.2.1`.

I've also made it a bit clearer which fields are available and which aren't:

```js
require('isomorphic-fetch');
const {default: ApolloClient, createNetworkInterface} = require('apollo-client');
const gql = require('graphql-tag').default;

const client = new ApolloClient({
  networkInterface: createNetworkInterface({uri: 'http://www.githunt.com/graphql'})
});

const query = gql`
fragment RepositoryData on Repository {
  owner {
    login
  }
}

{
  feed(type: NEW, limit: 5) {
    repository {
      ...RepositoryData
      name
      owner {
        avatar_url
      }
    }
  }
}
`;

const promise = client.query({query}).then(({data}) => {
  const {owner} = data.feed[0].repository;
  console.log(owner.avatar_url); // https://avatars…
  console.log(owner.login); // undefined
}, e => {
  console.error(e);
});

```

It seems like the problem is still there. If you remove the `owner` override of the query, the `login` is printed to the console.

**Comment by helfer** [comment]
Ok, thanks, I'll look into it before the end of the week!

**Comment by ksmth** [comment]
if you change
```
repository {
  ...RepositoryData
  name
  owner {
    avatar_url
  }
}
```

to read like this:
```
repository {
  name
  owner {
    avatar_url
  }
  ...RepositoryData
}
```
it works. This also works for as many fragments in any order, so it seems for now it's best to always put them last.

**Comment by helfer** [comment]
Thanks @ksmth, that's a pretty simple workaround then. I'll still try to get it fixed, but I might make it lower priority for now.

**Comment by cesarsolorzano** [comment]
I've found that this issue is related to graphql-anywhere, and in fact, it has already been reported: apollographql/graphql-anywhere#41

**Comment by stubailo** [comment]
Note that this issue is now fixed as of `graphql-anywhere` 3.1.0
